crates/payout_curve/src/lib.rs

```diff
@@ -0,0 +1,685 @@
+                    extra_precision: 0,
+                },
+            ))
+        })
+        .collect::<Result<Vec<(_, _)>>>()?;
+
+    Ok(pieces)
+}
+
+/// Calculates the lower range payout points between 0 and the lower liquidation point
+fn calculate_lower_range_payouts(
+    coordinator_direction: Direction,
+    total_collateral: u64,
+    liquidation_price_lower_bound: Decimal,
+) -> Result<(PayoutPoint, PayoutPoint)> {
+    let (lower, upper) = match coordinator_direction {
+        // if coordinator is short, he gets everything from 0 until the trader's liquidation point
+        Direction::Short => (
+            PayoutPoint {
+                event_outcome: 0,
+                outcome_payout: total_collateral,
+                extra_precision: 0,
+            },
+            PayoutPoint {
+                event_outcome: liquidation_price_lower_bound
+                    .to_u64()
+                    .expect("to be able to fit decimal into u64"),
+                outcome_payout: total_collateral,
+                extra_precision: 0,
+            },
+        ),
+        // if coordinator is long, he gets 0 from 0 until his liquidation point
+        Direction::Long => (
+            PayoutPoint {
+                event_outcome: 0,
+                outcome_payout: 0,
+                extra_precision: 0,
+            },
+            PayoutPoint {
+                event_outcome: liquidation_price_lower_bound
+                    .to_u64()
+                    .expect("to be able to fit decimal into u64"),
+                outcome_payout: 0,
+                extra_precision: 0,
+            },
+        ),
+    };
+
+    Ok((lower, upper))
+}
+
+/// Calculates the upper range payout points between upper liquidation point and BTCUSD_MAX_PRICE
+fn calculate_upper_range_payouts(
+    coordinator_direction: Direction,
+    total_collateral: u64,
+    last_payout_point: PayoutPoint,
+) -> Result<(PayoutPoint, PayoutPoint)> {
+    let (lower_range_lower, lower_range_upper) = match coordinator_direction {
+        // if coordinator is long, he gets everything from the trader's liquidation point to
+        // infinity
+        Direction::Long => (
+            PayoutPoint {
+                event_outcome: last_payout_point.event_outcome,
+                outcome_payout: last_payout_point.outcome_payout,
+                extra_precision: 0,
+            },
+            PayoutPoint {
+                event_outcome: BTCUSD_MAX_PRICE
+                    .to_u64()
+                    .expect("to be able to fit decimal into u64"),
+                outcome_payout: total_collateral,
+                extra_precision: 0,
+            },
+        ),
+        // if coordinator is short, he gets 0 from his liquidation point to infinity
+        Direction::Short => (
+            PayoutPoint {
+                event_outcome: last_payout_point.event_outcome,
+                outcome_payout: last_payout_point.outcome_payout,
```

Comment:
❓ Why is this not 0?

Reply:
Similar to above, we sometimes get a rounding error of 1 sat in the mid range, meaning this is `1` sat and not `0`.